src/languageservice/parser/ast-converter.ts

```diff
@@ -125,7 +126,9 @@ function convertSeq(node: YAMLSeq, parent: ASTNode, doc: Document, lineCounter:
 
 function convertScalar(node: Scalar, parent: ASTNode): ASTNode {
   if (node.value === null) {
-    return new NullASTNodeImpl(parent, node, ...toOffsetLength(node.range));
+    return ['null', '~'].includes(node.source?.toLowerCase())
+      ? new NullASTNodeImpl(parent, node, ...toOffsetLength(node.range))
+      : new NullObjectASTNodeImpl(parent, node, ...toOffsetLength(node.range));
```

Comment:
Introduce `Null Object` node **(type=object and value=null)** when no value instead of `Null node`